src/modules/TrackComponents/Block.ts

```diff
@@ -0,0 +1,56 @@
+import { Switch } from './Switch';
+
+export class Block {
+  // Failures (default to false)
+  public brokenRailFailure: boolean;
+  public signalFailure: boolean;
+  public engienFailure: boolean;
+
+  // Fields set by setter functions
+  public switch?: Switch;
+  public occupied: boolean;
+
+  /*
+   * TODO: other classes? beacon  and station
```

Comment:
@kjb135-coe @cwh35 Idk your plans for beacons, but this might be a good spot to implement it.